src/main/java/HttpServer/MockTraffic.java

```diff
@@ -0,0 +1,35 @@
+package HttpServer;
+
+import java.util.ArrayList;
+
+public class MockTraffic implements Readable {
+
+    private ArrayList<String> request;
+    private int index;
+
+    public MockTraffic() {
+    }
+
+    public void emulateSimpleCurl() {
```

Comment:
This has some smells of violating the OCP (open-closed principle), since adding new sample requests requires coming in and modifying this code.

I think it might be worth extracting some sample requests into their own files.

Reply:
We refactored MockTraffic to make it more extensible. The request is formulated right when we test. The tradeoff is we have a little bit of duplication across test files. Also the MockTraffic gets destroyed at the end, so we end up making several of them per test file. We're curious what you think.

Reply:
A couple points:
1. Small amounts of duplication across test files (with an emphasis on small), is not always bad. Oftentimes, one is making the tradeoff of increasing duplication for increased readability and simplicity.
2. Also, I'm not sure what you mean by making several `MockTraffic` per test file. Do you mean you are creating an instance of `MockTraffic` per test? If so, I'm not sure I see the issue. Is instantiating `MockTraffic` slow/expensive?